components/push/ffi/src/lib.rs

```diff
@@ -186,15 +184,15 @@ pub extern "C" fn push_dispatch_for_chid(
     log::debug!("push_dispatch_for_chid");
     MANAGER.call_with_result_mut(error, handle, |mgr| -> Result<String> {
         let chid = chid.as_str();
-        if let Some(record) = mgr.get_record_by_chid(chid)? {
-            let dispatch = json!({
-                "uaid": record.uaid,
-                "scope": record.scope,
-            });
-            Ok(dispatch.to_string())
-        } else {
-            // TODO: either Error or return Option
-            Ok(String::from(""))
+        match mgr.get_record_by_chid(chid)? {
+            Some(record) => {
+                let dispatch = json!({
+                    "uaid": record.uaid,
+                    "scope": record.scope,
+                });
+                Ok(dispatch.to_string())
+            }
+            None => Ok(String::from("")),
```

Comment:
We should really be handling None the way every other FFI layer handles it (e.g. as null pointer and not as an empty string)...

But we can get to that later.

Reply:
Good point. May try to address it here if it's not terrible.

Reply:
Hrm, not sure how to deal with the Option<Result<String>> in the `callRust` layer. Don't want to block either while I try to figure it out.

Reply:
Feel free to file a follow up and I'll try and get to it when I have time. I have a good idea of what's required and it's not totally trivial.

Reply:
filed #877